docs/orchestrating-elastic-stack-applications/stack-config-policy.asciidoc

```diff
@@ -0,0 +1,237 @@
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/put-snapshot-repo-api.html[Snapshot Repositories]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/slm-api-put-policy.html[Snapshot Lifecycle Policies]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/put-pipeline-api.html[Ingest pipelines]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/ilm-put-lifecycle.html[Index Lifecycle Policies]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-template.html[Index templates]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-component-template.html[Components templates]
+- link:https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html[Role mappings]
+
+A policy can be applied to all or a subset of Elasticsearch clusters in a given namespace or all namespaces managed by the operator.
+
+Settings described in a policy applied by the ECK operator are immutable through the Elasticsearch REST API.
+
+It is not allowed to configure an Elasticsearch cluster with more than one policy currently.
+
+[float]
+[id="{p}-{page_id}-definition"]
+== Define Elastic Stack configuration policies
+
+Elastic Stack configuration policies can be defined in an `StackConfigPolicy` resource. Each `StackConfigPolicy` must have the following fields:
+
+* `name` is a unique name used to identify the policy.
+* `spec.elasticsearch` describes the settings to configure, at least one setting must be defined. Each following field is a named entity map:
+  ** `clusterSettings` are the settings commonly found in elasticsearch.yaml settings.
+  ** `snapshotRepositories` are snapshot repositories, to define off-cluster storage location for your snapshots.
+  ** `snapshotLifecyclePolicies` are snapshot lifecycle policies, to automatically take snapshots and control how long they are retained.
+  ** `securityRoleMappings` are role mappings, to define which roles are assigned to each user by identifying them through rules.
+  ** `ingestPipelines` are ingest pipelines, to perform common transformations on your data before indexing.
+  ** `indexLifecyclePolicies` are index lifecycle policies, to automatically manage the index lifecycle.
+  ** `indexTemplates.componentTemplates` are component templates that are building blocks for constructing index templates that specify index mappings, settings, and aliases.
+  ** `indexTemplates.composableIndexTemplates` are index templates to define settings, mappings, and aliases that can be applied automatically to new indices.
+
+The following fields are optional:
+
+* `namespace` is the namespace of the `StackConfigPolicy` resource, it is used to identify the Elasticsearch clusters to which this policy applies. If it equals to the operator namespace, all namespaces managed by the operator are used, otherwise only the policy namespace is used.
```

Comment:
```suggestion
* `namespace` is the namespace of the `StackConfigPolicy` resource and used to identify the Elasticsearch clusters to which this policy applies. If it equals to the operator namespace, all namespaces managed by the operator are used, otherwise only the policy namespace is used.
```